src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -31,7 +31,13 @@ public TemporaryStorageServiceFactory()
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             var textFactory = workspaceServices.GetService<ITextFactoryService>();
-            return new TemporaryStorageService(textFactory);
+
+            // MemoryMapped files which are used by the TemporaryStorageService are not present
```

Comment:
They are present on Mono/OSX and we are using them in VSMac.

Reply:
@Therzok Thanks for taking a look. I updated so that we return the MemoryMapped backed service when running on Mono.

Reply:
Thanks!